src/query/sql/src/executor/physical_plans/physical_hash_join.rs

```diff
@@ -320,8 +320,26 @@ impl PhysicalPlanBuilder {
                 probe_fields.extend(build_fields);
                 probe_fields
             }
-            JoinType::LeftSemi | JoinType::LeftAnti => probe_fields,
-            JoinType::RightSemi | JoinType::RightAnti => build_fields,
+            JoinType::LeftSemi | JoinType::LeftAnti | JoinType::RightSemi | JoinType::RightAnti => {
+                let result_fields = if join.join_type == JoinType::LeftSemi
+                    || join.join_type == JoinType::LeftAnti
+                {
+                    probe_fields
+                } else {
+                    build_fields
+                };
+                for field in result_fields.iter() {
+                    match field.name().parse::<usize>() {
+                        Ok(index) if !column_projections.contains(&index) => {
+                            return Err(ErrorCode::SemanticError(
```

Comment:
Can we check it in the parse phase? @andylokandy 
Snow returns:
```
Syntax error: unexpected 'ANTI'. (line 6)
```

Reply:
It can only be checked after the bind phase.